File: Numbers And Math/1518. Water Bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int ans = numBottles; // Initialize the answer with the initial bottles
        
        while (numBottles >= numExchange) {
            int newBottles = numBottles / numExchange; // Bottles we can get after exchange
            ans += newBottles; // Add these new bottles to our answer
            numBottles = newBottles + (numBottles % numExchange); // Remaining bottles after exchange
        }
        
        return ans;
    }
};

File: Numbers And Math/1518. Water Bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int total=0;
        total+=numBottles;
        int newbot=0;
        while(numBottles>=numExchange)
        {
            newbot=numBottles/numExchange;
            total+= newbot;
            numBottles =newbot+(numBottles % numExchange);

        }
        return total;
    }
}